### PR TITLE
feat(provider/azure): Add availability zones support in server group …

### PR DESCRIPTION
**Pull request by michaeljqzq**
Add availability zones support in server group configuration

### CODE REVIEW

**Comment by michaeljqzq** [comment]
Linked with PR in `clouddriver` https://github.com/spinnaker/clouddriver/pull/3467

File: app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js [hunk 1]
```diff
@@ -164,5 +164,33 @@ describe('Service: azureServerGroupConfiguration', function() {
...
+        regionsSupportZones: [],
+        availabilityZones: ['1', '2', '3'],
+      };
+
+      service.configureZones(this.command);
+
+      expect(this.command.backingData.filtered.zones).toEqual([]);
+    });
+
+    it('returns actual zone list if region not supported', function() {
```
Comment (sbwsg) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js]: I think this description string is intended to end with "if region **is** supported".

In this test, `'eastasia'` is in the set of supported zones, right?

File: app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.js [hunk 2]
```diff
@@ -143,6 +143,22 @@ module.exports = angular
         return result;
       }
 
+      function configureZones(command) {
+        const result = { dirty: {} };
+        const filteredData = command.backingData.filtered;
+        if (command.region === null) {
```
Comment (sbwsg) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.js]: Should this also return early if `command.region` is `undefined`?  If so I suggest changing this to:

```
if (!command.region) {
  return result;
}
```

File: app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js [hunk 3]
```diff
@@ -164,5 +164,33 @@ describe('Service: azureServerGroupConfiguration', function() {
...
+        regionsSupportZones: ['eastasia'],
+        availabilityZones: ['1', '2', '3'],
+      };
+
+      service.configureZones(this.command);
+
+      expect(this.command.backingData.filtered.zones).toEqual(
+        this.command.backingData.credentialsKeyedByAccount[this.command.credentials].availabilityZones,
+      );
+    });
```
Comment (sbwsg) [inline app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js]: I suggest adding a test for behaviour expected when `this.command.region` is null or undefined.

File: app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js (hunk 1 above)
Comment (michaeljqzq) [reply app/scripts/modules/azure/serverGroup/configure/serverGroupConfiguration.service.spec.js]: you're right, made mistake here

**Comment by michaeljqzq** [comment]
> I have added a few suggestions. Once those are addressed or rejected I will merge.

Thanks @sbwsg , resolved these suggestions

**Pull request merged**